wolfssl/openssl/bn.h

```diff
@@ -85,7 +85,7 @@ WOLFSSL_API int wolfSSL_BN_is_zero(const WOLFSSL_BIGNUM*);
 WOLFSSL_API int wolfSSL_BN_is_one(const WOLFSSL_BIGNUM*);
 WOLFSSL_API int wolfSSL_BN_is_odd(const WOLFSSL_BIGNUM*);
 WOLFSSL_API int wolfSSL_BN_is_negative(const WOLFSSL_BIGNUM*);
-WOLFSSL_API int wolfSSL_BN_is_word(const WOLFSSL_BIGNUM*, WOLFSSL_BN_ULONG);
+WOLFSSL_API int wolfSSL_BN_is_word(const WOLFSSL_BIGNUM*, unsigned long);
```

Comment:
@douzzer , why not just fix `WOLFSSL_BN_ULONG` to be `unsigned long`? Also unsigned long will be different sizes depending on the architecture. Example: 16-bit micro unsigned long is 32-bit most others is 64-bit.

Reply:
Oh hey good idea, how 'bout that, `WOLFSSL_BN_ULONG` is only used in ssl.c/bn.h.  It had the sound of something used internally so I hadn't considered changing its meaning.  A quick survey suggests changing it from `mp_digit` to `unsigned long` would have exactly the result I'd intended.

Btw, notice that `wolfSSL_BN_get_word()` was already returning a target-native `unsigned long`.  My patch is just intended to harmonize the API with that.  The substantive change, of course, is that all of these routines now know how to convert multiple mp/sp digits to a single target-native unsigned long when it will fit, which is to say, they DTRT when `[ms]p_digit` is smaller than `unsigned long`, and only error when the MP/SP number can't fit in the native long word.

Re your second point, the purpose of the patch is indeed to arrange for these API wrapper functions to follow the target-native `unsigned long` word size, to meet caller expectations of OpenSSL semantics, and in particular, to not truncate 65537 with a coercion to 16 bit `unsigned short` in the 32 bit SP use case.

I'll push code shortly that restores the cosmetics of the bn.h API, while defining `WOLFSSL_BN_ULONG` as target-native `unsigned long` rather than `mp_digit`/`sp_digit`.